security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -201,21 +200,33 @@ func (sc *SecretCache) setRootCert(rootCert []byte, rootCertExpr time.Time) {
 	sc.rootCertMutex.Unlock()
 }
 
-// getRootCertInfo returns cached root cert and cert expiration time. This method is thread safe.
+// getRootCertInfo returns cached trust bundles and earliest cert expiration time. This method is thread safe.
 func (sc *SecretCache) getTrustBundles() (trustBundles map[string][]byte, earliestExpiry time.Time) {
 	sc.rootCertMutex.RLock()
 	trustBundles = sc.trustBundleMap
 	earliestExpiry = sc.trustBundleExpireTime
+	if trustBundles != nil && sc.rootCert != nil {
+		trustBundles[sc.configOptions.TrustDomain] = sc.rootCert
+		if sc.rootCertExpireTime.Before(earliestExpiry) {
+			earliestExpiry = sc.rootCertExpireTime
+		}
+	}
 	sc.rootCertMutex.RUnlock()
 	return trustBundles, earliestExpiry
 }
 
-// setRootCert sets root cert into cache. This method is thread safe.
-func (sc *SecretCache) setTrustBundles(trustBundles map[string][]byte, earliestExpiry time.Time) {
+// setRootCert stores trust bundles into the cache. This method is thread safe.
+func (sc *SecretCache) SetTrustBundles(trustBundles map[string][]byte, earliestExpiry time.Time) {
 	sc.rootCertMutex.Lock()
-	sc.trustBundleMap = trustBundles
-	sc.trustBundleExpireTime = earliestExpiry
+	trustBundlesUpdated := !compareTrustBundles(sc.trustBundleMap, trustBundles)
```

Comment:
probably not in the scope of this pr, but this function should better be called `trustBundlesEqual` or something like that. If trustBundle were a type, we could have a `func (tb TrustBundle) Equals(another TrustBundle)`.